app/add-details/page.tsx

```diff
@@ -0,0 +1,152 @@
+'use client';
+
+import { useState } from 'react';
+import { useRouter } from 'next/navigation';
+import { UUID } from 'crypto';
+import { updateUserPlants } from '@/api/supabase/queries/updateUserPlants';
+import PlantDetails from '@/components/PlantDetails';
+import { Plant, UserPlants } from '@/types/schema';
+
+export default function Home() {
+  const [currentIndex, setCurrentIndex] = useState<number>(1);
+  const [details, setDetails] = useState<Partial<UserPlants>[]>([]);
+  const [isDisabled, setIsDisabled] = useState<boolean[]>([true]);
+  const router = useRouter();
+
+  const plants: Plant[] = [
+    {
+      id: '43c19f80-8205-4d03-b323-05c220550bf0',
+      plant_name: 'cabbage',
+      us_state: 'string',
+      harvest_season: 'SPRING',
+      water_frequency: 'string',
+      weeding_frequency: 'string',
+      indoors_start: 'string',
+      indoors_end: 'string',
+      outdoors_start: 'string',
+      outdoors_end: 'string',
+      transplant_start: 'string',
+      transplant_end: 'string',
+      harvest_start: 'string',
+      harvest_end: 'string',
+      beginner_friendly: true,
+      plant_tips: 'string',
+      img: 'string',
+      difficulty_level: 'HARD',
+      sunlight_min_hours: 1,
+      sunlight_max_hours: 1,
+    },
+    {
+      id: '43c19f80-8205-4d03-b323-05c220550bf0',
+      plant_name: 'tomato',
+      us_state: 'string',
+      harvest_season: 'SPRING',
+      water_frequency: 'string',
+      weeding_frequency: 'string',
+      indoors_start: 'string',
+      indoors_end: 'string',
+      outdoors_start: 'string',
+      outdoors_end: 'string',
+      transplant_start: 'string',
+      transplant_end: 'string',
+      harvest_start: 'string',
+      harvest_end: 'string',
+      beginner_friendly: true,
+      plant_tips: 'string',
+      img: 'string',
+      difficulty_level: 'HARD',
+      sunlight_min_hours: 1,
+      sunlight_max_hours: 1,
+    },
+  ];
+  const user_id: UUID = 'e72af66d-7aae-45f6-935a-187197749d9f';
+  const getDefaultDate = () => new Date().toISOString().substring(0, 10);
+
+  function move(steps: number) {
+    // Navigate between plants and save input data
+    const currentDetail = details[currentIndex - 1];
+    if (
+      (!currentDetail || !currentDetail.date_added) &&
+      currentIndex <= plants.length
+    ) {
+      updateInput('date_added', getDefaultDate());
+    }
+    if (
+      steps > 0 &&
+      (!currentDetail?.planting_type ||
+        currentDetail?.planting_type == 'SELECT')
```

Comment:
bet bet i added the null hceck bc at the very start planting_type doesnt have a value
so like, lets say the user presses transplant and then decides to go back to select option at a later time, that would change the attribute from null -> transplant -> select. So just checking in case that ends up happening